src/core/include/openvino/core/model.hpp

```diff
@@ -24,9 +24,15 @@
 #include "openvino/runtime/tensor.hpp"
 
 namespace ov {
+namespace frontend {
+class FrontEnd;
+}
 class ModelAccessor;
 /// A user-defined function.
 class OPENVINO_API Model : public std::enable_shared_from_this<Model> {
```

Comment:
Do we need to have final for Model class?

Reply:
Yes, I believe we can do it in separate PR, if we agree to do this. Formally, making ngraph::Function as 'final' is backward-incompatible changes, so I wanted to track this separately. I briefly checked - with `final` keyword local build succeeded so at least there is no tests where we do deriving from ov::Model (ex ngraph::Function)